### PR TITLE
Lint tests, split up into individual files, allow to run tests standalone

### PR DESCRIPTION
**Pull request by SuperSandro2000**

### CODE REVIEW

File: test/source-any/05-immutability-regex-blocks-re-transfer.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - from: ${SOURCE_SPEC}
+      to: { container: ${CONTAINER_BASE}-test5 }
+      only: '/$|file.txt'
+      immutable: '.*.txt'
+EOF
+
+expect test5 <<-EOF
+>> just/another/file.txt
+Hello Another World.
+EOF
```
Comment (majewsky) [inline test/source-any/05-immutability-regex-blocks-re-transfer.sh:22]: Where is `just/another/file.txt` coming from? I don't see it being uploaded neither in `test/lib.sh` nor anywhere above this assertion in the same file.

Upon further inspection, other tests similarly reference files in the assertions that are never uploaded, so I'm not going to repeat this note again over there.

**Review by majewsky — CHANGES_REQUESTED** [review]
The CI has not attempted to run the tests, and I'm not sure why. In any case, the Makefile needs to be adjusted to run the new test harness (cf. definition of the `functional-test` target in `Makefile.maker.yaml`).

Also, I was not run the tests myself, with the following error message:
```
$ ./test/run.sh all
>> Preparing source container...
1632407606/just/some/files/1.txt
1632407606/just/some/files/2.txt
source-any/01-mirror-from-http.sh: line 4: cd: /x/src/github.com/sapcc/swift-http-import/test/test/..: No such file or directory
>> Cleaning up container swift-http-import-source...
1632407606/just/some/files/1.txt 1632407606/just/some/files/2.txt
```

Please let me know if I'm holding anything wrong. I'll leave the review comments that I accumulated while reading through roughly the first half of the code, but those are probably incomplete since I was not able to run the test suite yet.

File: test/source-any/05-immutability-regex-blocks-re-transfer.sh (hunk 1 above)
Comment (SuperSandro2000) [reply test/source-any/05-immutability-regex-blocks-re-transfer.sh:22]: Yeah, I didn't change this yet. Right now all tests still need to be executed in order. When they can be executed out of order it is also very easy to parallelise them.

**Comment by SuperSandro2000** [comment]
> Please let me know if I'm holding anything wrong.

After allowing the tests to be run standalone and fiddling with it around a bit I did not re-check ./tests all. Should work now ™️

**Review by majewsky — CHANGES_REQUESTED** [review]
The basic structure is looking good, but several tests have lost their meaning as a result of their disentanglement from the other tests' side effects. Please have another look at the tests to see if they make sense in the context of what they are supposedly testing. I've marked the cases where I saw contradictions and unclarities.

**Review by majewsky — APPROVED** [review]

**Pull request merged**